.github/workflows/publish_docker_images.yml

```diff
@@ -9,101 +9,101 @@ on:
         uses: docker/build-push-action@v4
         with:
           context: .
           file: ./osdk/tools/docker/Dockerfile
           platforms: linux/amd64
           push: true
-          load: true
-          tags: asterinas/osdk:${{ steps.fetch-versions.outputs.aster_version }}
+          tags: asterinas/osdk:${{ steps.setup-check.outputs.image_version }}
           build-args: |
-            ASTER_RUST_VERSION=${{ steps.fetch-versions.outputs.rust_version }}
+            ASTER_RUST_VERSION=${{ steps.setup-check.outputs.rust_version }}
+
+  push_osdk_tdx_image:
+    needs: push_osdk_image
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup and Check
+        uses: ./.github/actions/docker
+        id: setup-check
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          token: ${{ secrets.DOCKERHUB_TOKEN }}
+          image_name: 'osdk-tdx'
 
       - name: Build and push the OSDK development image for Intel TDX
-        if: env.osdk_tdx_image_exists == 'false'
+        if: ${{ steps.setup-check.outputs.is_existed == 'false' }}
         uses: docker/build-push-action@v4
         with:
           context: .
           file: ./osdk/tools/docker/tdx/Dockerfile
           platforms: linux/amd64
           push: true
-          tags: asterinas/osdk:${{ steps.fetch-versions.outputs.aster_version }}-tdx
+          tags: asterinas/osdk:${{ steps.setup-check.outputs.image_version }}-tdx
           build-args: |
-            BASE_VERSION=${{ steps.fetch-versions.outputs.aster_version }}
+            BASE_VERSION=${{ steps.setup-check.outputs.image_version }}
+
+  push_asterinas_image:
+    needs: push_osdk_image
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup and Check
+        uses: ./.github/actions/docker
+        id: setup-check
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          token: ${{ secrets.DOCKERHUB_TOKEN }}
+          image_name: 'asterinas'
 
       - name: Build and push the Asterinas development image
-        if: env.asterinas_image_exists == 'false'
+        if: ${{ steps.setup-check.outputs.is_existed == 'false' }}
         uses: docker/build-push-action@v4
         with:
           context: .
           file: ./tools/docker/Dockerfile
           platforms: linux/amd64
           push: true
-          load: true
-          tags: asterinas/asterinas:${{ steps.fetch-versions.outputs.aster_version }}
+          tags: asterinas/asterinas:${{ steps.setup-check.outputs.image_version }}
           build-args: |
-            BASE_VERSION=${{ steps.fetch-versions.outputs.aster_version }}
+            BASE_VERSION=${{ steps.setup-check.outputs.image_version }}
+
+  push_asterinas_tdx_image:
+    needs: push_asterinas_image
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup and Check
+        uses: ./.github/actions/docker
```

Comment:
I just realized that our current practice, which is invoking an _action_, is actually a bad practice. The intention (reusing code) is good and it does achieve it. But it is done in a very cumbersome way.

A much simpler approach to avoiding repeating ourselves in the context of Github workflows is to leverage this [`jobs.<job_id>.steps[*].run`](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsrun) feature. It allows us to put reusable steps in a script.

If you agree with my "discovery", then let's replace the existing actions with scripts. And this PR should also add a script and invoke the script in the workflow YML file.